lib/fluent/plugin/kafka_producer_ext.rb

```diff
@@ -38,9 +38,15 @@ def produce_for_buffered(value, key: nil, topic:, partition: nil, partition_key:
 end
 
 # for out_kafka2
+# Majority (if not all) of this code is lifted from https://github.com/zendesk/ruby-kafka/blob/master/lib/kafka/producer.rb
+# with the main difference where we have removed any checks regarding max_buffer_bytesize and max_buffer_size
+# The reason for doing this is to provide a better UX for our users where they only need to set those bounds in
+# the Buffer section using `chunk_limit_size` and `chunk_limit_records`.
+#
+# We should reconsider this in the future in case the `ruby-kafka` library drastically changes its internal.
```

Comment:
This is what I was able to infer from the intent of this file, thought I should document it here so future us will know what this is doing. Let me know if this is incorrect.

Reply:
It seems correct.
@repeatedly Please comment on this if you have any concerns.

Reply:
This comment is correct. I implemented this class to remove some overhead/unmatched parts from ruby-kafka's producer.

Reply:
Thanks for your comment!